Tests/UInt128Tests/UInt128Tests.swift

```diff
@@ -478,10 +478,17 @@ class BinaryIntegerTests : XCTestCase {
     }
 
     func testInitFloatingPoint() {
+        #if arch(x86_64)
         var tests = [(input: Float80(), result: UInt128())]
         tests.append((input: Float80(0.1), result: UInt128()))
         tests.append((input: Float80(1.0), result: UInt128(1)))
         tests.append((input: Float80(UInt64.max), result: UInt128(UInt64.max)))
+        #else
+        var tests = [(input: 0.0, result: UInt128())]
+        tests.append((input: 0.1, result: UInt128()))
+        tests.append((input: 1.0, result: UInt128(1)))
+        tests.append((input: Double(UInt32.max), result: UInt128(UInt32.max)))
+        #endif
```

Comment:
Is this something that affects M1 processors? If so, could we perhaps introduce a test to make sure we never introduce this issue again?

Reply:
🤔 I think the [`Float80`](https://developer.apple.com/documentation/swift/float80) is not available on:

1. iOS devices (arm/arm64)
2. Mac with the Apple Silicon (arm64)
3. Apple Watch with armv7k CPU (32bit)

So I simply workaround this by `#if arch(...)`. However the case `UInt64.max` is lost.

Reply:
Interesting. Is there anything equivalent or larger for any of those platforms that you are aware of?

Reply:
Yes. Need new Decimal conversion functions and new test cases for the conversion. 😂

Reply:
If you're not up for that right now, we can just stick with what you have.